blocklylib-core/src/main/java/com/google/blockly/android/ui/fieldview/BasicFieldVariableView.java

```diff
@@ -199,10 +200,36 @@ public void run() {
          * @param variableNameManager The name manager containing the variables.
          * @param context A context for inflating layouts.
          * @param resource The {@link TextView} layout to use when inflating items.
+         * @param mSpinner Spinner this is being used in
          */
         public VariableViewAdapter(Context context, NameManager variableNameManager,
-                                   @LayoutRes int resource) {
-            super(context, resource);
+                                   @LayoutRes int resource, AppCompatSpinner mSpinner) {
+            super(context, resource, mSpinner);
+
+            mVariableNameManager = variableNameManager;
+            mVars = mVariableNameManager.getUsedNames();
+
+            mRenameString = context.getString(R.string.rename_variable);
+            mDeleteString = context.getString(R.string.delete_variable);
+            refreshVariables();
+            variableNameManager.registerObserver(new DataSetObserver() {
+                @Override
+                public void onChanged() {
+                    refreshVariables();
+                }
+            });
+        }
+
+        /**
+         * This constructor is for compatibility reasons, or for whe it is not in a Spinner.
```

Comment:
When might the adapter be used outside a spinner? Do we really need this constructor, or is this just for code backward compatibility for third party code?

If every use should include the snipper, and this original version is only here to transition older code, add a debug only log message (`BuildConfig.DEBUG`). Additionally, add a comment with an expected removal date; one year is good.

Reply:
Test code, and third party compatibility.